### DISCUSSION TITLE
fmincon failed

### DISCUSSION THREAD

**Discussion by TYY126** [yalmip/YALMIP#1348] [Possible bug or missing functionality] [answered]
Why is my function fmincon showing failed
![image](https://github.com/yalmip/YALMIP/assets/134586099/fcd19947-be68-4fde-9fa6-8de90f8c3ae9)

**Comment by johanlofberg** [comment]
You will have to setup some trivial problem and solve it using fmincon, with 'debug' turned on, to see what happens in the call

**Reply by TYY126** [reply]
![image](https://github.com/yalmip/YALMIP/assets/134586099/e22122f3-4425-4b60-980a-520bb552f653)
![image](https://github.com/yalmip/YALMIP/assets/134586099/e64f5b3a-022e-4fb7-845b-ddec40b07955)

**Reply by johanlofberg** [reply]
You have to use fmincon of course. Using gurobi won't reveal anything about why fmincon fails

**Reply by TYY126** [reply]
![image](https://github.com/yalmip/YALMIP/assets/134586099/e97ae288-00af-4414-8c82-947961fcfceb)
![image](https://github.com/yalmip/YALMIP/assets/134586099/f3c2765c-c8b5-4cf9-a573-218ad20bda3c)
What should I do

**Reply by johanlofberg** [reply]
Precisely what fmincon is telling you to do in the error message.

**Reply by TYY126** [reply]
Am I going to download fmincon separately？

**Reply by johanlofberg** [reply]
As it says, you don't have the optimization toolbox installed, or you have a broken installation. Completely unrelated to yalmip

**Reply by TYY126** [reply]
So what should I do now

**Reply by johanlofberg** [reply, marked as answer]
fmincon does not work as the optimization toolbox isn't installed. Ergo, the optimization toolbox must be installed.

**Reply by TYY126** [reply]
Thank you very much for your reply. I now know how to do it
